### PR TITLE
feat(cmd): Adding fail-non-empty flag6132

### PR DESCRIPTION
**Pull request by Ronnie-personal**
### Why the changes in this PR are needed? 
This PR addresses issue #6132 

### What are the changes in this PR? 
'--fail-non-empty' flag is added to 'opa exec' CLI.

### CODE REVIEW

File: cmd/exec_test.go [hunk 1]
```diff
@@ -326,13 +367,116 @@ func TestFailFlagCases(t *testing.T) {
...
+		}`,
+			},
+			decision:    "fail/defined/flag/fail_test",
+			expectError: false,
+			expected: util.MustUnmarshalJSON([]byte(`{"result": [{
+			"path": "/files/test.json",
+			"result": []
+		}]}`)),
+			failNonEmpty: true,
+		},
```
Comment (anderseknert) [inline cmd/exec_test.go:479]: It would be good to add a test case for an empty set coming from a partial rule, i.e. something like:

```rego
package fail.non.empty.flag

fail_test[message] {
    false
    message := "not gonna happen"
}
```

File: cmd/exec_test.go [hunk 2]
```diff
@@ -326,13 +367,116 @@ func TestFailFlagCases(t *testing.T) {
...
+			"path": "/files/test.json",
+			"result": ["hello"]
+		}]}`)),
+			failNonEmpty: true,
+		},
+		{
+			description: "--fail-non-empty with true boolean result",
+			files: map[string]string{
+				"files/test.json": `{"foo": 7}`,
+				"bundle/x.rego": `package fail.defined.flag
```
Comment (anderseknert) [inline cmd/exec_test.go:423]: These package names are misleading. Maybe use something like `fail.non.empty.flag` instead?

**Review by anderseknert — COMMENTED** [review]
This is a great start! I left a few comments, but this is looking good :)

**Comment by Ronnie-personal** [comment]
> This is a great start! I left a few comments, but this is looking good :)

Thank you for the review,  I will check them out.

**Comment by Ronnie-personal** [comment]
Could you help me to understand the failure at https://github.com/open-policy-agent/opa/actions/runs/5897177383/job/15996426056?pr=6153#step:6:258?
Do we have the failure details in the log?
 I run 'make test' in my local environment, opa/runtime actually passed.
 
 Thanks.

File: cmd/exec.go [hunk 3]
```diff
@@ -68,8 +68,9 @@ e.g., opa exec --decision /foo/bar/baz ...`,
 	addConfigOverrides(cmd.Flags(), &params.ConfigOverrides)
 	addConfigOverrideFiles(cmd.Flags(), &params.ConfigOverrideFiles)
 	cmd.Flags().StringVarP(&params.Decision, "decision", "", "", "set decision to evaluate")
-	cmd.Flags().BoolVarP(&params.FailDefined, "fail-defined", "", false, "exits with non-zero exit code on defined/non-empty result and errors")
-	cmd.Flags().BoolVarP(&params.Fail, "fail", "", false, "exits with non-zero exit code on undefined/empty result and errors")
+	cmd.Flags().BoolVarP(&params.FailDefined, "fail-defined", "", false, "exits with non-zero exit code on defined result and errors")
+	cmd.Flags().BoolVarP(&params.Fail, "fail", "", false, "exits with non-zero exit code on undefined result and errors")
+	cmd.Flags().BoolVarP(&params.FailNonEmpty, "fail-non-empty", "", false, "exits with non-zero exit code on non empty result and errors")
```
Comment (anderseknert) [inline cmd/exec.go:73]: Nit: but we say "non-zero", so we should say "non-empty" too.

**Review by anderseknert — COMMENTED** [review]
Looks good to me! Just a nit to fix :)

I'll test this out on Monday and if it's all good we should have this merged.

File: cmd/exec.go (hunk 3 above)
Comment (Ronnie-personal) [reply cmd/exec.go]: Thanks for the comment, I have modified it  at this commit https://github.com/open-policy-agent/opa/pull/6153/commits/f91f52d206c6c74157faf88bf1d7966f75f45649

**Comment by anderseknert** [comment]
I've now tried this out locally and it seems like it's working as expected. Thanks for you work on this @Ronnie-personal 👍

**Review by anderseknert — APPROVED** [review]
